### PR TITLE
QA Incorporation

### PR DESCRIPTION
**Pull request by oscarso2000**
incomplete as of Dec 13 12am ET

### CODE REVIEW

**Comment by oscarso2000** [comment]
do not merge, just code to look through in the future if needed

File: app/q_and_a/cosine.py [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+
+def remove_punctuation(data):
+    symbols = "!\"#$%&()*+-./:;<=>?@[\]^_`{|}~\n"
+    for i in range(len(symbols)):
+        data = np.char.replace(data, symbols[i], ' ')
+        data = np.char.replace(data, "  ", " ")
+    data = np.char.replace(data, ',', '')
+    return data
+
+def remove_apostrophe(data):
```
Comment (TrueshotBarrage) [inline app/q_and_a/cosine.py:72]: Does this need to be a separate function?

File: client/src/components/ResultsList.jsx [hunk 2]
```diff
@@ -12,22 +12,21 @@ import { outline } from '../actions';
...
 
-export const ResultsList = ({ results, screenshots }) => {
+export const ResultsList = ({ results, screenshots, QA }) => {
     const ResultsArray = [];
-    const len = results.length;
+    const len = results.length;    
 
     for (let i = 0; i < len; i++) {
-        // console.log(results[i])
         ResultsArray.push(<Result data={results[i]} screenshots={screenshots} />);
```
Comment (TrueshotBarrage) [inline client/src/components/ResultsList.jsx:20]: Does this need to have a key specified for each item in the list?

**Comment by TrueshotBarrage** [comment]
welp realized this is not supposed to ever be pushed lmao

**Pull request closed without merging**